### PR TITLE
Make the list of banned users configurable  Draft pull

### PR DESCRIPTION
**Pull request by jeffiscow2**

### CODE REVIEW

File: devel/ansible/roles/dev/files/config.toml [hunk 1]
```diff
@@ -67,6 +67,24 @@ issuer_origin = "http://badges.tinystage.test"
 issuer_url = "http://fedoraproject.org"
 issuer_email = "badges@fedoraproject.org"
 
+# Users skipped when awarding badges
+skip_users = [
```
Comment (abompard) [inline devel/ansible/roles/dev/files/config.toml]: This list should be added before `[consumer_config.cache]` in the configuration file in order for it to end up in the main configuration directory, here it'll go in the `badge_issuer` subdirectory.

File: fedbadges/rules.py [hunk 2]
```diff
@@ -222,8 +191,8 @@ def _get_candidates(self, msg: Message, tahrir: TahrirDatabase):
         # Remove None
         candidates = frozenset([e for e in candidates if e is not None])
 
-        # Exclude banned usernames
-        candidates = candidates.difference(self.banned_usernames)
+        # Exclude skipped usernames
+        candidates = candidates.difference(self.skip_users)
```
Comment (abompard) [inline fedbadges/rules.py]: Here, `self.skip_users` is the filename `"config.toml"`. We want it to be the actual list of usernames.

You can get that list by using the `self.config.get("skip_users", [])`.

**Review by abompard — CHANGES_REQUESTED** [review]
Thanks, please also add the list to the `fedbadges.toml.example` file.

**Comment by abompard** [comment]
Also, please squash your commits.

**Comment by jeffiscow2** [comment]
I Cant get the commits to squash and I've tried a million things. I pretty shure I made it worse... lol

**Comment by abompard** [comment]
OK I think I fixed the squashes :-)

File: fedbadges/rules.py [hunk 3]
```diff
@@ -219,12 +186,16 @@ def _get_candidates(self, msg: Message, tahrir: TahrirDatabase):
...
+                candidates = frozenset(candidates - self.config.get("skip_users"))
+            else:
+                continue
+
         # Remove None
         candidates = frozenset([e for e in candidates if e is not None])
 
-        # Exclude banned usernames
-        candidates = candidates.difference(self.banned_usernames)
-
```
Comment (abompard) [inline fedbadges/rules.py]: You don't really need to add this loop, you could reuse the previous "difference" line. You just need to convert the datastructure that comes from the configuration file as a list into a set:
```python
candidates = candidates.difference(frozenset(self.config.get("skip_users", [])))
```

File: fedbadges.toml.example [hunk 4]
```diff
@@ -53,6 +53,25 @@ fasjson_base_url = "https://fasjson.fedoraproject.org"
...
+    "bodhi",
+    "taskotron",
+    "pagure",
+    "packit",
+    "packit-stg",
+    "koschei",
+    "distrobuildsync-eln/jenkins-continuous-infra.apps.ci.centos.org",
+    "osbuild-automation-bot",
+    "zodbot",
+    "root",
```
Comment (abompard) [inline fedbadges.toml.example:69]: Looks like there are less entries in this list than in the one that was in the `rules.py` file initially.

**Review by abompard — CHANGES_REQUESTED** [review]

File: fedbadges.toml.example (hunk 4 above)
Comment (jeffiscow2) [reply fedbadges.toml.example:69]: ope added the rest of the skip_users

**Review by jeffiscow2 — COMMENTED** [review]
I turned rules reload back to 15 from 1

File: .vscode/settings.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (abompard) [inline .vscode/settings.json]: This file is unrelated to this PR, please don't add it.

File: fedbadges.toml.example [hunk 6]
```diff
@@ -52,11 +52,9 @@ fasjson_base_url = "https://fasjson.fedoraproject.org"
 
 # Check for new rules every these many minutes
 rules_reload_interval = 15
-
-# Users skipped when awarding badges
-skip_users = [
+# Users skped when awarding badges
```
Comment (abompard) [inline fedbadges.toml.example]: Typo here.

File: fedbadges/rules.py [hunk 7]
```diff
@@ -187,13 +187,10 @@ def _get_candidates(self, msg: Message, tahrir: TahrirDatabase):
...
         # Skipped usernames
-        for candidates in self.config.get("skip_users"):
-            if candidates == self.config.get("skip_users"):
-                candidates = frozenset(candidates - self.config.get("skip_users"))
-            else:
-                continue
+        candidates = candidates.difference(frozenset(self.config.get("skip_users", [])))
 
         # Remove None
+
```
Comment (abompard) [inline fedbadges/rules.py]: You don't really need to add this empty line, comments are usually right above what they are commenting.

**Review by abompard — CHANGES_REQUESTED** [review]

**Comment by abompard** [comment]
Looks like you've merged your local copy into the changed that I had made to this pull request, which added back all the commits including merge conflicts.
I'm going to push my changes again. Then you can do `git reset --hard origin/develop` and it'll bring your local copy in line with the PR. I'll incorporate the changes you did in ff8787f.

**Comment by abompard** [comment]
The problem may be coming from the fact that the Git integration in VS Code is really basic and doesn't let you do advanced stuff without an additional plugin. In any case, please remember to check the "Files changed" section of a PR after you've pushed, to make sure the changes are exactly what you want in the PR and nothing else.

**Pull request merged**